mscore/startupWizard.cpp

```diff
@@ -0,0 +1,81 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//  Copyright (C) 2017 Werner Schweer and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+
+#include "startupWizard.h"
+#include "musescore.h"
+#include "libmscore/mscore.h"
+#include "shortcut.h"
+
+namespace Ms {
+QString StartupWizard::layoutInput()
+      {
+      QStringList layoutList;
+      QMap<QString, QString> layoutToShortcut = getShortcutMap();
+      QList<QString> keyboardLayouts = layoutToShortcut.keys();
+      for (auto layout : keyboardLayouts) {
+            layoutList.append(qApp->translate("keyboard-layout", layout.toStdString().c_str()));
+      }
+
+      QString labelMessage = QObject::tr("Enter your keyboard layout");
+      QString result = QInputDialog::getItem(new QWidget(), "Startup Wizard", labelMessage, layoutList);
```

Comment:
No. Let's keep it simple, it's the goal here, to make MuseScore more simple, not more complex. The first time we start MuseScore, we detect the system language and we propose what we think is the best suited keyboard layout and then the best suited shortcut list. I think we should have a single list of keyboard layout. If the language is fr_FR then choose french France azerty in the list and so default shortcut will be this, but the list should be complete : It's not because my language is fr_FR that I don't have a american qwerty plugged.